sql/src/main/java/io/crate/execution/dml/upsert/ShardUpsertRequest.java

```diff
@@ -367,7 +371,15 @@ public void writeTo(StreamOutput out, @Nullable Streamer[] insertValueStreamers)
             } else {
                 out.writeVInt(0);
             }
-            out.writeLong(version);
+            // Version is long but we write it as integer here. This was introduced by a previous bug.
+            // We chose to keep protocol backwards-compatibility for rolling upgrades.
+            // Crate > 3.0 has this fixed.
+            if (version > Integer.MAX_VALUE) {
+                throw new IllegalStateException(String.format(Locale.ENGLISH,
```

Comment:
exceptions raised on streaming are hard to track as they mostly not bubbling up including a stacktrace. Also if one item inside the request has this problem, all others will also fail as the whole request is failing. I'd suggest doing this check at an upperlevel so we could ensure that the exception is somehow populated and maybe logged. also other items would not suffer from this failure.